pkg/processor/broadcast_storage_handler.go

```diff
@@ -82,6 +84,14 @@ func (h *BroadcastStorageHandler) TransactionConfirmed(
 		return fmt.Errorf("%w: confirmed transaction did not match intent", err)
 	}
 
+	// Validate destination memo if it's needed
+	if intentMemo, found := intentMetadata["memo"]; found {
+		observedMemo := transaction.Metadata["memo"]
```

Comment:
do we need to check if `memo` exists in `transaction.Metadata`?

Reply:
I think so because it can:
* Gate nil pointer for the following code
* Better for performance because if `memo` doesn't exist we don't go through comparison code/logic, sometimes the metadata could be complicated but currently we only consider `memo` so additional comparison is not necessary if `memo` doesn't exist
* Easy to maintain

Reply:
I am referring to Line 89: `observedMemo := transaction.Metadata["memo"]`
Why we do not add Gate to this

Reply:
Yup, we should add existence check for it as well and throw `memo not found` error, good catch!